web/api/diff.py

```diff
@@ -1,390 +1,577 @@
-        for model, modeldict in self.children.items():
-            for identity, reldict in modeldict.items():
-                if reldict['side'] != 'both':
-                    self.rel_changed = True
-
-        # Mark as cleaned.
-        self.cleaned = True
-
-    def add_property(self, key, value, side):
-        """Add property to the node.
-
-        :param key: Key|Name of the property
-        :type key: str
-        :param value: str
-        :type value: str
-        :param side: Which side is the property coming from.
-        :type side: str
+        skip = (page - 1) * self.pagesize
+        q = '{} \nSKIP {}\nLIMIT {}'.format(str(self), skip, self.pagesize)
+        rows = []
+        for record in self._fetch(q):
+            row = OrderedDict()
+            for select in self.selects:
+                row[select] = record[select]
+            rows.append(row)
+        return rows
+
+    def fetch_all(self):
+        """Fetch all pages.
+
+        :yields: One result per match.
+        :ytype: OrderedDict
         """
-        if side == LEFT:
-            this = self.left_props
-        else:
-            this = self.right_props
-        this[key] = value
+        page = 1
+        go = True
+        while go:
+            rows = self.fetch(page)
+            for row in rows:
+                yield row
+            if not rows:
+                go = False
+            page += 1
 
-    def add_child(self, model, node, side):
-        """Add child to the node.
 
-        :param model: Type|Label of the child node.
-        :type model: str
-        :param node: Node to add
-        :type node: Node
-        :param side: Which side is the node coming from?
-        :type side: str
+class DiffSideQuery(DiffQuery):
+    """Models a query to find subpaths.
+
+    Finds all paths on a 'side'. That is all paths of nodes that belong
+    to one time but not the second time.
+
+    This is used to find nodes that have been added or deleted.
+    """
+    def _match_clause(self):
+        """Build the multistage match clause.
+
+        Find all paths that match in t2. Then find all paths that match in t1
+        that are not in t2.
+
+        :returns: Major portion of cipher query
+        :rtype: str
         """
-        model_rel = self.children.setdefault(model, {})
-        node_rel = model_rel.setdefault(node.identity, {
-            'side': side,
-            'node': node
-        })
-        if side != node_rel['side']:
-            node_rel['side'] = BOTH
-
-    def remove_child(self, model, identity):
-        """Remove a child.
-
-        :param model: Type|Label of the node to remove
-        :type model: str
-        :param identity: Identity of the node
-        :type identity: str
+        path_vars = []
+        for model, _ in self.full_path:
+            path_vars.append(self._var_from_label(model))
+        path_vars.append(self._var_from_label(self.end))
+
+        cipher = 'MATCH p_t2 = '
```

Comment:
imo, this section could use a comment example. It's pretty hard to navigate because it's broken up so much.

Reply:
Added query examples for each clause.